ios/mapbox_maps_flutter/Sources/mapbox_maps_flutter/Classes/MapRecorderController.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+@_spi(Experimental) import MapboxMaps
+import Flutter
+
+/// Controller for MapRecorder functionality.
+///
+/// Provides functions to record and replay API calls of a MapboxMap instance.
+/// These recordings can be used to debug issues which require multiple steps to reproduce.
+/// Additionally, playbacks can be used for performance testing custom scenarios.
+class MapRecorderController: _MapRecorderMessenger {
+    private let mapboxMap: MapboxMap
+    private var recorder: MapRecorder?
```

Comment:
Same as Android - the recorder instance is never cleaned up, potentially causing memory leaks.

Add an explicit cleanup:
```swift
deinit {
    recorder = nil
}
```